spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -301,24 +303,34 @@ private Dataset<FileURI> actualFileIdentDS() {
 
   private Dataset<String> listedFileDS() {
     List<String> subDirs = Lists.newArrayList();
-    List<String> matchingFiles = Lists.newArrayList();
+    Set<String> matchingFiles = Sets.newHashSet();
 
     Predicate<FileStatus> predicate = file -> file.getModificationTime() < olderThanTimestamp;
     PathFilter pathFilter = PartitionAwareHiddenPathFilter.forSpecs(table.specs());
 
+    List<String> locationsToList = Lists.newArrayList();
+    if (location.equals(table.location())) {
+      locationsToList.add(dataLocation());
+      locationsToList.add(metadataFileLocation());
+    } else {
+      locationsToList.add(location);
+    }
```

Comment:
So if I'm reading this correctly, this will not check the table location for orphan files, I think that might be reasonable.

Reply:
This is the existing behavior to check the given location  instead of table location, when specified via `org.apache.iceberg.actions.DeleteOrphanFiles#location`
On a related note, the current code doesn't check
`org.apache.iceberg.TableProperties#WRITE_DATA_LOCATION`,
`org.apache.iceberg.TableProperties#WRITE_METADATA_LOCATION`

which this PR check.

Reply:
What if users have modified data and metadata location table properties multiple times? In that case orphan files from old location will never be cleaned up?

Reply:
@ajantha-bhat I think that is the current behavior as well?